euphonic/spectra/base.py

```diff
@@ -180,9 +174,9 @@ def _cut_x_ticks(x_tick_labels: XTickLabels | None,
         """Crop and shift x labels to new x range"""
         if x_tick_labels is None:
             return None
-        else:
-            return [(int(x - x0), label) for (x, label) in x_tick_labels
-                    if (x >= x0) and ((x1 is None) or (x < x1))]
+
+        return [(int(x - x0), label) for (x, label) in x_tick_labels
+                if (x >= x0) and ((x1 is None) or (x < x1))]
```

Comment:
Comment describing logic of filtering? Also, is the `x` of `x_tick_labels` not `int`?

`# Check x in range(x0, x1)`

Could also say `x1 = np.inf if x1 is None else x1` and just have
`x0 <= x < x1`

Reply:
How about
```python
    @staticmethod
    def _cut_x_ticks(x_tick_labels: XTickLabels | None,
                     x0: int,
                     x1: int | None) -> XTickLabels | None:
        """Crop x labels to new x range, shifting indices accordingly"""
        if x_tick_labels is None:
            return None

        def in_range(x: int) -> bool:
            return x0 <= x < (float('inf') if x1 is None else x1)

        return [(int(x - x0), label)
                for (x, label) in x_tick_labels if in_range(x)]
```

Reply:
Ok will leave for now and have created an Issue to look at this separately

Reply:
Sorry, I missed that, was in a meeting. 

It looks ok, but I think it would be fine to just have:
```python
    @staticmethod
    def _cut_x_ticks(x_tick_labels: XTickLabels | None,
                     x0: int,
                     x1: int | None) -> XTickLabels | None:
        """Crop x labels to new x range, shifting indices accordingly"""
        if x_tick_labels is None:
            return None

        if x1 is None:
            x1 = float('inf')

        return [(int(x - x0), label)
                for (x, label) in x_tick_labels if x0 <= x < x1]
```